openedx/features/redhouse_panel/api/v0/users_account_stats.py

```diff
@@ -0,0 +1,82 @@
+from django.contrib.auth import get_user_model
+from django.contrib.sites.models import Site
+from django.db.models import F, Q, Count, Case, When, OuterRef, Exists, BooleanField
+from django.shortcuts import get_object_or_404
+from rest_framework import status
+from rest_framework.authentication import SessionAuthentication
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from rest_framework_oauth.authentication import OAuth2Authentication
+
+from student.models import CourseAccessRole
+from student.roles import CourseStaffRole, CourseInstructorRole
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from openedx.core.lib.api.permissions import IsStaffOrOwner
+
+User = get_user_model()
+
+
+class UsersAccountStatsAPIView(APIView):
+    authentication_classes = (
+        JwtAuthentication,
+        OAuth2Authentication,
+        SessionAuthentication,
+    )
+    permission_classes = (IsAuthenticated, IsStaffOrOwner,)
+
+    default_stats = {
+        'instructor_count': 0,
+        'student_count': 0
+    }
+
+    def get(self, request, pk):
+        site = get_object_or_404(Site, pk=pk)
+
+        _stats = get_user_model().objects.filter(
```

Comment:
Please try to simplify your query It's too complicated right now. I can't understand it. Try to break it down in chunks.